### PR TITLE
Redo documentation

### LINKED ISSUE #45: Redo documentation
The README is partially wrong and some features need to be documented:
* The http-proxy is *not* a "real" HTTP proxy, the ruby example needs to be removed and replaced with maybe a curl example
* Behaviour with `Authorization` header set needs to be explained better (multiple clients)
* API v8/v6 images should no longer be mentioned, latest supports all versions as long as the Path exists in twilight
* Metrics shouldn't be documented as "Grafana metrics", rather as "Prometheus metrics" since Grafana also supports e.g. InfluxDB as a backend
* After tackling #26, document this behavior properly

### PR DESCRIPTION
**Pull request by Gelbpunkt**
Entire revamp of the documentation as explained in the linked issue.

[Rendered](https://github.com/Gelbpunkt/http-proxy/blob/redo-documentation/README.md)

Closes #45

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,12 +3,32 @@
 `http-proxy` is a ratelimited HTTP proxy in front of the Discord API, making use
 of [twilight].
 
-### Using it
+## Using it
```
Comment (7596ff) [inline README.md]:
```suggestion
## Use
```

File: README.md [hunk 2]
```diff
@@ -3,12 +3,32 @@
...
 
-HTTP clients often support proxies, such as Ruby's [`Net::HTTP`]. Read into your
-HTTP client to see how to use it.
+This proxy is not a "real" HTTP `CONNECT` TCP proxy, instead it mocks to be the
+Discord API.
```
Comment (7596ff) [inline README.md]:
```suggestion
This proxy is not a "real" HTTP `CONNECT` TCP proxy, it instead mocks the
Discord API.
```

File: README.md [hunk 3]
```diff
@@ -3,12 +3,32 @@
...
 
-`twilight_http` natively supports using `twilight_http_proxy`, so you can use it like
+In order to use it, change the base URL of all your routes from
```
Comment (7596ff) [inline README.md]: I think for this section, we should prefer to begin with explaining the simplest use case, which I understand to be `curl http://localhost:3000/users/@me`. We can then mention that the user can add `/v9/` or another version string to the URL, and be clear whether or not the `/api/` segment is required or optional.

File: README.md [hunk 4]
```diff
@@ -39,15 +81,16 @@ Prebuilt Docker images are published on [Docker Hub].
 ```sh
 $ docker run -itd -e DISCORD_TOKEN="my token" -p 3000:80 twilightrs/http-proxy
 # Or with metrics enabled
-$ docker run -itd -e DISCORD_TOKEN="my token" -p 3000:80 twilightrs/http-proxy:metrics
+$ docker run -itd -e DISCORD_TOKEN="my token" -p 3000:80
+twilightrs/http-proxy:metrics
```
Comment (7596ff) [inline README.md]: was this meant to be wrapped?

**Review by 7596ff — CHANGES_REQUESTED** [review]

**Comment by Gelbpunkt** [comment]
Updated, hope I didn't miss anything

**Review by 7596ff — APPROVED** [review]

**Review by Erk- — APPROVED** [review]

**Pull request merged**